### ISSUE TITLE
Compiler warning about ABI of current target

### ISSUE THREAD

**Issue by carolynzech** [model-checking/kani#3878]
Upgrading our toolchain to 2/7/25, then running `./scripts/kani-regression.sh` now produces this warning:

```
   Compiling kani_core v0.59.0 (/Users/cmzech/kani/library/kani_core)
warning: target feature `neon` must be enabled to ensure that the ABI of the current target can be implemented correctly                                                                                                                                                                
  |
  = note: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #116344 <https://github.com/rust-lang/rust/issues/116344>


warning: target feature `neon` must be enabled to ensure that the ABI of the current target can be implemented correctly                                                                                                                                                                
  |
  = note: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #116344 <https://github.com/rust-lang/rust/issues/116344>


   Compiling std v0.59.0 (/Users/cmzech/kani/library/std)
warning: target feature `neon` must be enabled to ensure that the ABI of the current target can be implemented correctly                                                                                                                                                                
  |
  = note: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #116344 <https://github.com/rust-lang/rust/issues/116344>

```

The solution may just be to enable this target feature, but we need to dig more into the linked issue to ensure that's the right course of action.

**Comment by tautschnig** [comment]
I think we should just enable that `target_feature` to silence the warning: Kani does not care about the ABI for it uses its own interpretation of MIR and never deals with the actual object code generated for the target platform.

**Comment by zhassan-aws** [comment]
These are the warnings on a Ubuntu machine:
```
warning: target feature `x87` must be enabled to ensure that the ABI of the current target can be implemented correctly
  |
  = note: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #116344 <https://github.com/rust-lang/rust/issues/116344>


warning: target feature `sse2` must be enabled to ensure that the ABI of the current target can be implemented correctly
  |
  = note: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #116344 <https://github.com/rust-lang/rust/issues/116344>
```
